packages/wallets/src/components/Base/WalletPasswordField/WalletPasswordField.tsx

```diff
@@ -0,0 +1,79 @@
+import React, { FC, useState } from 'react';
+import { zxcvbn } from '@zxcvbn-ts/core';
+import { WalletTextField } from '../WalletTextField';
+import './WalletPasswordField.scss';
+
+type StrengthMessage = Record<1 | 2 | 3 | 4, string>;
+
+interface WalletPasswordFieldProps {
+    label?: string;
+    maxLength?: number;
+    messageObj?: StrengthMessage;
+}
+
+const WalletPasswordField: FC<WalletPasswordFieldProps> = ({ label = 'Password', maxLength, messageObj }) => {
+    const [password, setPassword] = useState<string>('');
+    const hasMessage = !!messageObj;
+    const hasMaxLength = !!maxLength;
+
+    /**
+     * StrengthMessage is a mapping of zxcvbn score to corresponding strength levels.
+     * 1 = Weak
+     * 2 = Moderate
+     * 3 = Strong
+     * 4 = Complete
+     *
+     * @example
+     * const messageObj = {
+     *    1: 'Weak',
+     *    2: 'Moderate',
+     *    3: 'Strong',
+     *    4: 'Complete',
+     * };
+     *
+     * @param {number} strength - The zxcvbn password strength score.
+     * @returns {string} A string representing the password strength level.
+     */
+
+    const getPasswordStrength = () => {
+        const result = zxcvbn(password);
+        return result.score;
+    };
+
+    const strength = getPasswordStrength();
+
+    const getProgressColor = () => {
+        const strengthColors = [
+            'wallets-password__meter--initial',
+            'wallets-password__meter--weak',
+            'wallets-password__meter--moderate',
+            'wallets-password__meter--strong',
+            'wallets-password__meter--complete',
+        ];
+
+        return strengthColors[strength];
+    };
```

Comment:
Maybe separate `strengthColors` out of component and just use `strengthColors[strength]` in place of `getProgressColor()`?